clientconn.go

```diff
@@ -563,16 +578,44 @@ func (cc *ClientConn) scWatcher() {
 // We should never need to replace an addrConn with a new one. This function is only used
 // as newAddrConn to create new addrConn.
 // TODO rename this function and clean up the code.
-func (cc *ClientConn) resetAddrConn(addr Address, block bool, tearDownErr error) error {
+func (cc *ClientConn) resetAddrConn(addrs []Address, block bool, tearDownErr error) error {
+	// if current transport in addrs, just change lists to update order and new addresses
+	// not work for roundrobin
+	cc.mu.Lock()
+	if len(cc.conns) != 0 && (reflect.TypeOf(cc.dopts.balancer) == reflect.TypeOf(&pickFirst{})) {
+		var currentAc *addrConn
+		for _, v := range cc.conns {
+			currentAc = v
+			break
+		}
+		var addrInUse bool
```

Comment:
Consider moving the rest of this logic to a method in addrConn (UpdateAddresses).